File: tournaments/checkPalindrome/checkPalindrome.js
function checkPalindrome(inputString) {
    const n = inputString.length;
    const leftHalf = inputString.substring(0, ~~(n / 2));
    const rightHalf = [...inputString.substring(Math.ceil(n / 2))].reverse().join('');

    return leftHalf === rightHalf;
}

File: tournaments/checkPalindrome/checkPalindrome.cpp
bool checkPalindrome(std::string inputString) {

  bool isPalindrome = true;
  for (int i = 0; i < inputString.size(); i++) {
    if (inputString[i] != inputString[(int)inputString.size() - i - 1]) {
      isPalindrome = false;
      break;
    }
  }

  return isPalindrome;
}
